src/PackageCompiler.jl

```diff
@@ -212,25 +214,40 @@ function create_sysimg_object_file(object_file::String, packages::Vector{String}
     precompile_code = """
         # This @eval prevents symbols from being put into Main
         @eval Module() begin
+
+            using Logging
+            # Pass through the user's logging level to the spawned process.
+            logger = ConsoleLogger(stderr, Logging.$(Logging.min_enabled_level(global_logger())))
+            global_logger(logger)
+
             PrecompileStagingArea = Module()
             for (_pkgid, _mod) in Base.loaded_modules
                 if !(_pkgid.name in ("Main", "Core", "Base"))
                     eval(PrecompileStagingArea, :(const \$(Symbol(_mod)) = \$_mod))
                 end
             end
-            precompile_statements = String[]
-            $precompile_statements
-            for statement in sort(precompile_statements)
-                # println(statement)
-                # The compiler has problem caching signatures with `Vararg{?, N}`. Replacing
-                # N with a large number seems to work around it.
-                statement = replace(statement, r"Vararg{(.*?), N} where N" => s"Vararg{\1, 100}")
-                try
-                    Base.include_string(PrecompileStagingArea, statement)
-                catch
-                    # See julia issue #28808
-                    @debug "failed to execute \$statement"
+            let (failures, errors) = (0, 0)
+                precompile_statements = String[]
+                $precompile_statements
+                for statement in sort(precompile_statements)
+                    # println(statement)
+                    # The compiler has problem caching signatures with `Vararg{?, N}`. Replacing
+                    # N with a large number seems to work around it.
+                    statement = replace(statement, r"Vararg{(.*?), N} where N" => s"Vararg{\1, 100}")
+                    try
+                        success = Base.include_string(PrecompileStagingArea, statement)
+                        if !success
+                            failures += 1
+                            @debug "Precompilation failed: \$statement"
+                        end
+                    catch e
+                        # See julia issue #28808
+                        errors += 1
+                        @debug "Error executing \$statement:\n\$e"
+                    end
                 end
+                total = length(precompile_statements)
+                @info "PackageCompiler: completed \$total precompile statements with \$failures failures, \$errors errors."
```

Comment:
One problem here is that people think there is something wrong with their code when they see "x errors" but there is really nothing they can do to fix it. So I'm a little bit against adding this as something people see by default.

Reply:
Yeah, i had the same concern, and wasn't sure what to do about it.

Maybe a flag, to control reporting precompilation success rate?

Or we could print this to `@debug`, too, but debug is very noisy and i think we'd like to see this all the time.

Reply:
Do you have any suggestions? :) Thanks!

Reply:
Maybe we can also print the link to the julia issue that's in the comment, above?

Or do you think I should just move this to a `@debug` log as well?